File: solutions_5706278382862336_1/Python/Ummae/A_part_elf.py
import sys
from fractions import gcd
rl = lambda: sys.stdin.readline().strip()


def get_simple_frac(a, b):
    while True:
        c = gcd(a, b)
        if c >= 2:
            a /= c
            b /= c
        else:
            break
    return a, b


def is_2pow(n):
    while True:
        if (n & 1) != 0:
            return False

        n /= 2
        if n == 1:
            break
    return True


T = int(rl())
for tcase in range(T):
    P, Q = map(int, rl().split('/'))
    P, Q = get_simple_frac(P, Q)
    ans = 1
    possible = True
    if not is_2pow(Q) and not (P == 1 and Q == 1):
        possible = False
    else:
        for i in range(1, 41):
            n = pow(2, i)
            if float(P) / float(Q) >= 1.0 / float(n):
                break
            else:
                ans += 1

    if possible:
        print 'Case #%d: %d' % (tcase + 1, ans)
    else:
        print 'Case #%d: %s' % (tcase + 1, 'impossible')

File: solutions_5706278382862336_0/C++/gawarkiewicz/A - Part Elf.cpp
#include <iostream>
#include <sstream>
#include <string>
using namespace std;

long long gcd(long long a, long long b)
{
	while (b != 0)
	{
		long long r = a % b;
		a = b;
		b = r;
	}
	return a;
}

int main() {
	int T;
	cin >> T;
	for (int q = 1; q <= T; ++q) {
		string s;
		cin >> s;
		for (auto& a : s) if (a == '/') a = ' ';
		stringstream ss(s);
		long long P, Q;
		ss >> P >> Q;
		auto cd = gcd(P, Q);
		P /= cd;
		Q /= cd;
		bool possible = true;
		for (long long n = Q; n > 2; n /= 2) {
			if (n % 2 != 0) possible = false;
		}
		if (Q < P) possible = false;
		if (!possible) {
			cout << "Case #" << q << ": impossible" << endl;
			continue;
		}
		while (P > 1) {
			P /= 2;
			Q /= 2;
		}
		int ans = 0;
		while (Q > 1) {
			Q /= 2;
			++ans;
		}
		cout << "Case #" << q << ": " << ans << endl;
	}
	return 0;
}